BrightFutures/ExecutionContext.swift

```diff
@@ -50,9 +50,22 @@ public func toContext(queue: dispatch_queue_t) -> ExecutionContext {
     return Queue(queue: queue).context
 }
 
-typealias ThreadingModel = () -> ExecutionContext
+/// Defines BrightFutures' default threading behaviour, by default this is
+/// set to `StandardThreadingModel.AsynchronousOnCurrentThread`
+public var DefaultThreadingModel: StandardThreadingModel = .AsynchronousOnCurrentThread
```

Comment:
@matthew-healy @Thomvis Guys, I'm open to discussions about it and I think it's great that they are having place.

Though, I sincerely hate when library limits my capabilities. Yes, have default something for standard needs is great, but make a way to tweak things for advanced use cases.

In my situation I just need to have this functionality, because for my use case the threading model should dynamically be calculated depending on the current situation and will have much more advanced logic (specific for my particular needs).